src/hooks/use-import-export.tsx

```diff
@@ -0,0 +1,217 @@
+import * as Sentry from '@sentry/electron/renderer';
+import { __ } from '@wordpress/i18n';
+import { createContext, useMemo, useState, useCallback, useContext } from 'react';
+import { getIpcApi } from '../lib/get-ipc-api';
+import { ExportEvents } from '../lib/import-export/export/events';
+import { generateBackupFilename } from '../lib/import-export/export/generate-backup-filename';
+import { BackupCreateProgressEventData, ExportOptions } from '../lib/import-export/export/types';
+import { ImportExportEventData } from '../lib/import-export/handle-events';
+import { useIpcListener } from './use-ipc-listener';
+
+type ProgressState = {
+	[ siteId: string ]: {
+		statusMessage: string;
+		progress: number;
+	};
+};
+
+interface ImportExportContext {
+	importState: ProgressState;
+	exportState: ProgressState;
+	exportFullSite: ( selectedSite: SiteDetails ) => Promise< void >;
+	exportDatabase: ( selectedSite: SiteDetails ) => Promise< void >;
+}
+
+const DEFAULT_STATE = {
+	statusMessage: __( 'Starting export...' ),
+	progress: 5,
+};
+
+const ImportExportContext = createContext< ImportExportContext >( {
+	importState: {},
+	exportState: {},
+	exportFullSite: async () => undefined,
+	exportDatabase: async () => undefined,
+} );
+
+export const ImportExportProvider = ( { children }: { children: React.ReactNode } ) => {
+	const [ importState, setImportState ] = useState< ProgressState >( {} );
+	const [ exportState, setExportState ] = useState< ProgressState >( {} );
+
+	const exportSite = useCallback(
+		async ( selectedSite: SiteDetails, options: ExportOptions ) => {
+			if ( exportState[ selectedSite.id ] ) {
+				return;
+			}
+
+			setExportState( ( prevState ) => ( {
+				...prevState,
+				[ selectedSite.id ]: DEFAULT_STATE,
+			} ) );
+
+			try {
+				await getIpcApi().exportSite( options, selectedSite.id );
+				getIpcApi().showNotification( {
+					title: selectedSite.name,
+					body: __( 'Export completed' ),
+				} );
```

Comment:
When exporting a site with a small file size, I noticed the progress bar is shown so quickly that it's barely seen. I'm thinking of delaying the resolution in favor of showing a complete export message in the progress bar.

Reply:
This will be addressed in https://github.com/Automattic/studio/pull/401.